src/placement/jump_map.c

```diff
@@ -541,6 +543,18 @@ get_target(struct pool_domain *root_pos, struct pool_domain *curr_pd, struct poo
 			range_set = isset_range(dom_cur_grp_used, start_dom, end_dom);
 			if (range_set) {
 				if (top == -1) {
+					if (curr_pd != root_pos) {
+						/* all domains within the PD have been used by the
+						 * current group, try to avoid co-located shards for
+						 * same group by ignoring the PD restrict.
+						 */
+						D_DEBUG(DB_PL, "PD[%d] all doms used for group, "
+							"weak the PD restrict\n",
+							(int)(curr_dom - root_pos));
+						curr_pd = root_pos;
+						curr_dom = curr_pd;
```

Comment:
not sure I follow here, will this cause dead loop if all PDs have been used up by all shards?

Reply:
I think it won't cause dead loop.
If all PDs all used up, it will go with original code at below L563 (reset_dom_cur_grp) that possibly cause shards co-locate.